crates/oxc_linter/src/lint_adapter.rs

```diff
@@ -715,15 +706,15 @@ impl<'a> Adapter<'a> for LintAdapter<'a> {
 
     fn resolve_neighbors(
         &self,
-        contexts: ContextIterator<'a, Self::Vertex>,
+        contexts: ContextIterator<'b, Self::Vertex>,
         type_name: &Arc<str>,
         edge_name: &Arc<str>,
         _parameters: &EdgeParameters,
         _resolve_info: &ResolveEdgeInfo,
-    ) -> ContextOutcomeIterator<'a, Self::Vertex, VertexIterator<'a, Self::Vertex>> {
+    ) -> ContextOutcomeIterator<'b, Self::Vertex, VertexIterator<'b, Self::Vertex>> {
         let res = match (type_name.as_ref(), edge_name.as_ref()) {
-            ("File", "ast_node") => resolve_neighbors_with(contexts, |v| {
-                v.as_file().unwrap().iter.borrow_mut().take().unwrap()
+            ("File", "ast_node") => resolve_neighbors_with(contexts, |_| {
+                Box::new((*self).semantic.nodes().iter().map(|x| Vertex::AstNode(*x)))
```

Comment:
The functionality with the `.clone()` we were trying to eliminate with @Boshen and @u9g is now here, and sans `.clone()`.